radixdlt-engine/src/main/java/com/radixdlt/atommodel/tokens/Bucket.java

```diff
@@ -16,30 +16,16 @@
  *
  */
 
-package com.radixdlt.constraintmachine;
+package com.radixdlt.atommodel.tokens;
 
-import com.radixdlt.atom.TxAction;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.identifiers.REAddr;
 
-import java.util.Objects;
-
-/**
- * Action which was parsed by constraint machine from a transaction
- */
-public final class REParsedAction {
-	private final TxAction txAction;
-
-	private REParsedAction(TxAction txAction) {
-		this.txAction = txAction;
-	}
-
-	public static REParsedAction create(
-		TxAction txAction
-	) {
-		Objects.requireNonNull(txAction);
-		return new REParsedAction(txAction);
-	}
-
-	public TxAction getTxAction() {
-		return txAction;
-	}
+// TODO: these methods are really here for client api,
+// TODO: fix to be more in line with scrypto abstractions
+public interface Bucket {
+	REAddr resourceAddr();
+	REAddr getOwner();
+	ECPublicKey getValidatorKey();
```

Comment:
I don't know how practical it's going to be, but it would be awesome if we can introduce another address type for validator public key before mainnet. Validator should be addressable in scrypto (in future).

Reply:
Actually the reason why I didn't give those REAddr is that I don't think those validator keys are actually addresses. I would imagine them to be more like keys in a map maintained by the SystemStaking component

Reply:
As discussed offline, the main reason for using single component is to maintain a global list of validators (without iterating searching the ledger state) and the top 100. Under such organisation, validator public key is just an internal key of the system staking component which exposes all staking functionality.